### PR TITLE
Add an Asyncify option to propagate the addList

### PR DESCRIPTION
**Pull request by nokotan**
This PR will take #3794 over.

New asyncify flag `--pass-arg=asyncify-propagate-addlist` will change `--pass-arg=asyncify-addlist` behavior, callers of functions in a asyncify-addlist will be also instrumented.

### CODE REVIEW

File: src/passes/Asyncify.cpp [hunk 1]
```diff
@@ -671,6 +679,30 @@ class ModuleAnalyzer {
...
+            if (verbose && !info.canChangeState) {
+              std::cout << "[asyncify] " << func->name
+                        << " is in the add-list, add\n";
+            }
+            info.canChangeState = true;
+            info.addedFromList = true;
+          }
+        }
+      }
+    }
```
Comment (kripken) [inline src/passes/Asyncify.cpp:704]: This duplicated code could be moved to a helper function, or at least a lambda.

File: test/lit/passes/asyncify_pass-arg=asyncify-addlist@foo_pass-arg=asyncify-removelist@foo.wast [hunk 2]
```diff
@@ -0,0 +1,199 @@
+;; NOTE: Assertions have been generated by update_lit_checks.py --all-items and should not be edited.
+;; NOTE: This test was ported using port_passes_tests_to_lit.py and could be cleaned up.
+
+;; RUN: foreach %s %t wasm-opt --asyncify --pass-arg=asyncify-addlist@foo -S --pass-arg=asyncify-removelist@foo -o - | filecheck %s
```
Comment (kripken) [inline test/lit/passes/asyncify_pass-arg=asyncify-addlist@foo_pass-arg=asyncify-removelist@foo.wast:4]: This file doesn't use the new flag - what is it testing in this PR?

File: src/passes/Asyncify.cpp (hunk 1 above)
Comment (nokotan) [reply src/passes/Asyncify.cpp:708]: done.

File: test/lit/passes/asyncify_pass-arg=asyncify-addlist@foo_pass-arg=asyncify-removelist@foo.wast (hunk 2 above)
Comment (nokotan) [reply test/lit/passes/asyncify_pass-arg=asyncify-addlist@foo_pass-arg=asyncify-removelist@foo.wast:4]: As discussed in https://github.com/emscripten-core/emscripten/issues/13150, this test will check the behavior when a function is matched in both addlist and removelist. I don't know how emitted warnings are ckecked in lit tests.

Comment (nokotan) [reply test/lit/passes/asyncify_pass-arg=asyncify-addlist@foo_pass-arg=asyncify-removelist@foo.wast]: Seems that test/unit/test_asyncify.py is suitable place for this test, adding test_asyncify_addlist_and_removelist to this file.

**Review by kripken — APPROVED** [review]
Sorry for the delay here! I think I missed a notification here or something. This looks great.

**Pull request merged**